lib/graphql/auth-guards.ts

```diff
@@ -0,0 +1,16 @@
+import { Role } from '@prisma/client';
+import { Resolver } from './resolvers';
+
+export const authenticate =
```

Comment:
Please add some documentation 😄

Reply:
```suggestion
export const authorize =
```
Authentication means login. Authorization means checking for roles and access to certain features, so I think this should be named `authorize`.

Reply:
nice catch, had it like this since i was originally planning on authenticating in the wrapper and not in context